Backpack/CardButton/Classes/CardButtonViewConfigurator.swift

```diff
@@ -0,0 +1,50 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Hold shared logic between BPKSaveCardButton & BPKShareCardButton
+final class CardButtonViewConfigurator {
+
+    // CardButton Required a larger hit area of at least 48px
+    let widthHeight = BPKSpacingLg * 2
+
+    func createContainedBackgroundCircle() -> UIView {
+        let circle = UIView()
+        circle.backgroundColor = BPKColor.textDisabledOnDarkColor.withAlphaComponent(0.5)
+        circle.isUserInteractionEnabled = false
+        circle.translatesAutoresizingMaskIntoConstraints = false
+        return circle
+    }
+
+    func shouldAddButtonBackground(style: BPKCardButtonStyle) -> Bool {
+        switch style {
+        case .contained:
+            return true
+        case .default, .onDark:
+            return false
+        }
+    }
+
+    func buttonBackgroundSize(size: BPKCardButtonSize) -> CGFloat {
+        switch size {
+        case .default:
+            return BPKIcon.concreteSizeForLargeIcon.width + BPKSpacingBase
+        case .small:
+            return BPKIcon.concreteSizeForSmallIcon.width + BPKSpacingBase
+        }
```

Comment:
The spacing around the icon is 8px, that would be BPKSpacingMd I believe.
BPKSpacingBase is 16px.

Reply:
Your are right. The spacing is 8px on the left & right. 

So, the width of the circle will be: 
`icon width + 8[left]  + 8[right]`


Should I rewrite it to be more verbose ?
`return BPKIcon.concreteSizeForLargeIcon.width + BPKSpacingMd + BPKSpacingMd`

Reply:
Ah! you're totally right 🤦. Seems I needed more coffee.
Leaving it as is is ok :D